packages/components/src/SvgIcon/utils/createSvgIcon.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from "react";
+import { $Diff } from "utility-types";
+import SvgIcon from "../SvgIcon";
+import type { SvgIconProps } from "../SvgIcon";
+
+type IconProps = $Diff<
+  SvgIconProps,
+  {
+    children: any;
+  }
+>;
+
+/**
+ * Generates SVG Icons from paths and fragments.
+ */
+export default function createSvgIcon(
+  path: React.ReactNode,
+  displayName: string,
+): React.FC<IconProps> {
+  const Component = (props: IconProps, ref: any) => {
+    return (
+      <SvgIcon ref={ref} {...props}>
+        {path}
+      </SvgIcon>
+    );
+  };
+
+  if (process.env.NODE_ENV !== "production") {
+    // Need to set `displayName` on the inner component for React.memo.
+    // React prior to 16.14 ignores `displayName` on the wrapper.
+    Component.displayName = `${displayName}Icon`;
+  }
+
+  return React.memo(React.forwardRef(Component));
```

Comment:
For our use case here do we need React.memo at all? Wondering if we can get rid of it (and the associated displayName stuff).

It's not clear that memoizing a leaf node like this matters too much.

(basically these icons are at the bottom of the React element tree. There isn't anything else to render other than an HTML element. So a memo/cache hit doesn't obviously save a bunch of rendering).

Reply:
@dcwither thoughts?